### PR TITLE
Fix contract compilation/deploy/testing issues

### PR DESCRIPTION
**Pull request by tsully**
## Changes

- Fix contract compilation issues
- Fix broken deploy script
- Add ganache to hardhat config
- Add sample .env file
- Add local dev instructions in README

### CODE REVIEW

File: scripts/init.js [hunk 1]
```diff
@@ -53,7 +52,7 @@ async function main() {
   await defaultPeerRewardsInstaller.deployed();
   console.log("[CONTRACT DEPLOYED] DefaultMembersInstaller: ", defaultPeerRewardsInstaller.address);
 
-  await defaultOSFactory.setOS(defaultOS.address)
+  await defaultOSFactory.setOS("default-id",defaultOS.address)
```
Comment (tsully) [inline scripts/init.js:55]: What os ID do we want to use here? Not sure if the OS ID is intended to be human readable.

File: test/modules/Mining.test.js [hunk 2]
```diff
@@ -32,221 +32,270 @@ describe("Mining.sol", function () {
...
-      await incrementWeek()
-      await this.epoch.incrementEpoch()
+      await incrementWeek();
+      await this.epoch.incrementEpoch();
       await expect(this.mining.issueRewards())
         .to.emit(this.mining, "RewardsIssued")
         .withArgs(
-          this.default.address, this.dev.address, await this.epoch.current(), 10 * MULT, await this.epoch.TOKEN_BONUS()
+          this.default.address,
+          this.vault.address,
```
Comment (tsully) [inline test/modules/Mining.test.js:187]: Sorry, I accidentally committed changes from my formatter. The only substantive change in this file was adding this `this.vault.address` argument.

**Comment by tsully** [comment]
On a related note, we really need to set up CI for this project...

File: test/modules/Mining.test.js (hunk 2 above)
Comment (zaqk) [reply test/modules/Mining.test.js:187]: agreed

**Review by fullyallocated — COMMENTED** [review]
Looks good, thanks @tsully for the fixes

**Pull request merged**